src/pybind/mgr/dashboard/frontend/src/app/shared/api/osd.service.ts

```diff
@@ -5,6 +5,111 @@ import { I18n } from '@ngx-translate/i18n-polyfill';
 
 import { ApiModule } from './api.module';
 
+export interface SmartAttribute {
+  flags: object;
+  id: number;
+  name: string;
```

Comment:
Really admire the effort of having a typed/structured front-end, but I'm still wondering if it make sense to take so much care for typing the front-end when we're allowing the data cross through our back-end unvalidated. I sometimes feel that we are filling our front-end of back-end concerns (not bad if the back-end cares too).

Reply:
I think both is useful, typed/structured front-end and a validated backend. I don't think we should wait for the back-end to implement typed/structured front-end, though.